test/unit/logic/multisignature.js

```diff
@@ -49,46 +53,566 @@ var multiSigAccount2 = {
+
+			expect(multisignature.calculateFee(trs).toString()).to.equal('2500000000');
+		});
+
+		it('should return correct fee based on formula for 8 keysgroup', function () {
+			trs.asset.multisignature.keysgroup = new Array(8).fill('+' + node.lisk.crypto.getKeys(node.randomPassword()).publicKey);
+
+			expect(multisignature.calculateFee(trs).toString()).to.equal('4500000000');
+		});
+
+		it('should return correct fee based on formula for 16 keysgroup', function () {
+			trs.asset.multisignature.keysgroup = new Array(16).fill('+' + node.lisk.crypto.getKeys(node.randomPassword()).publicKey);
+
+			expect(multisignature.calculateFee(trs).toString()).to.equal('8500000000');
+		});
+	});
+
+	describe('verify', function () {
+
+		describe('from multisignature.verify tests', function () {
+
+			it('should return error when min value is smaller than minimum acceptable value', function (done) {
+				var min = constants.multisigConstraints.min.minimum - 1;
+				var trs	= node.lisk.multisignature.createMultisignature(node.gAccount.password, null, ['+' + multiSigAccount1.publicKey, '+' + multiSigAccount2.publicKey], 1, 1);
+				trs.asset.multisignature.min = min;
+
+				multisignature.verify(trs, node.gAccount, function (err) {
+					expect(err).to.equal('Invalid multisignature min. Must be between 1 and 15');
+					done();
+				});
+			});
+		});
+
+		it('should return error when min value is greater than maximum acceptable value', function (done) {
+			var min = constants.multisigConstraints.min.maximum + 1;
+			var trs	= node.lisk.multisignature.createMultisignature(node.gAccount.password, null, ['+' + multiSigAccount1.publicKey, '+' + multiSigAccount2.publicKey], 1, min);
+
+			multisignature.verify(trs, node.gAccount, function (err) {
+				expect(err).to.equal('Invalid multisignature min. Must be between 1 and 15');
+				done();
+			});
+		});
+
+		describe('when asset = undefined', function () {
+
+			it('should call callback with error = "Invalid transaction asset"', function (done) {
+				delete trs.asset;
+
+				multisignature.verify(trs, sender, function (err) {
+					expect(err).to.equal('Invalid transaction asset');
+					done();
+				});
+			});
+		});
+
+		describe('when asset.multisignature = undefined', function () {
+
+			it('should call callback with error = "Invalid transaction asset"', function (done) {
+				delete trs.asset.multisignature;
+
+				multisignature.verify(trs, sender, function (err) {
+					expect(err).to.equal('Invalid transaction asset');
+					done();
+				});
+			});
+		});
+
+		describe('when asset.multisignature = []', function () {
+
+			it('should call callback with error = "Invalid multisignature keysgroup. Must not be empty"', function (done) {
+				trs.asset.multisignature.keysgroup = [];
+
+				multisignature.verify(trs, sender, function (err) {
+					expect(err).to.equal('Invalid multisignature keysgroup. Must not be empty');
+					done();
+				});
+			});
+		});
+
+		describe('when min <= 1', function () {
```

Comment:
Imo we should test also first and last valid path of a range - so 2 and 15 here.

Reply:
Added tests for these values